components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_accounts.go

```diff
@@ -51,37 +73,42 @@ func fetchAccounts(
 	connectorID models.ConnectorID,
 	ingester ingestion.Ingester,
 	scheduler task.Scheduler,
-) error {
+	state fetchAccountsState,
+) (fetchAccountsState, error) {
 	for page := 0; ; page++ {
-		pagedAccounts, err := client.GetAccounts(ctx, page, config.PageSize)
+		pagedAccounts, err := client.GetAccounts(ctx, page, config.PageSize, state.LastCreatedDate)
 		if err != nil {
-			return err
+			return state, err
 		}
 
 		if len(pagedAccounts.Content) == 0 {
 			break
 		}
 
 		if err := ingestAccountsBatch(ctx, connectorID, ingester, pagedAccounts.Content); err != nil {
-			return err
+			return state, err
 		}
 
 		for _, account := range pagedAccounts.Content {
+			if err := state.SetLatest(account); err != nil {
+				return state, err
+			}
```

Comment:
Also, usually when passing the last created at to the psp call, it will return all accounts with createat >= last created at, so you will process everytime the accounts with last created at

Reply:
In that case, we might add another method to the state, in order to perform the comparison on our side and avoid processing the same thing twice.